standard/basic-concepts.md

```diff
@@ -82,12 +82,14 @@ There are several different types of declaration spaces, as described in the fol
-- Additional local variable declaration spaces may occur within member declarations, anonymous functions and local functions. Names are introduced into these declaration spaces through *declaration_expression*s and *declaration_statement*s. Local variable declaration spaces may be nested, but it is an error for a local variable declaration space and a nested local variable declaration space to contain elements with the same name. Thus, within a nested declaration space it is not possible to declare a local variable, local function or constant with the same name as a parameter, type parameter, local variable, local function or constant in an enclosing declaration space. It is possible for two declaration spaces to contain elements with the same name as long as neither declaration space contains the other. Local declaration spaces are created by the following constructs:
+- Additional local variable declaration spaces may occur within member declarations, anonymous functions and local functions. Names are introduced into these declaration spaces through *pattern*s, *declaration_expression*s,  *declaration_statement*s and *exception_specifier*s. Local variable declaration spaces may be nested, but it is an error for a local variable declaration space and a nested local variable declaration space to contain elements with the same name. Thus, within a nested declaration space it is not possible to declare a local variable, local function or constant with the same name as a parameter, type parameter, local variable, local function or constant in an enclosing declaration space. It is possible for two declaration spaces to contain elements with the same name as long as neither declaration space contains the other. Local declaration spaces are created by the following constructs:
   - Each *variable_initializer* in a field and property declaration introduces its own local variable declaration space, that is not nested within any other local variable declaration space.
   - The body of a function member, anonymous function, or local function, if any, creates a local variable declaration space that is considered to be nested within the function’s local variable declaration space.
   - Each *constructor_initializer* creates a local variable declaration space nested within the instance constructor declaration. The local variable declaration space for the constructor body is in turn nested within this local variable declaration space.
-  - Each *block*, *switch_block*, *iteration_statement* and *using_statement* creates a nested local variable declaration space.
+  - Each *block*, *switch_block*, *specific_catch_clause*, *iteration_statement* and *using_statement* creates a nested local variable declaration space.
```

Comment:
Line 85 above now includes *exception_specifier* as introducing a declaration space, here *specific_catch_clause* is added. In the current grammar the former only occurs as part of the latter. Is the case here meant to be the *block* of a *specific_catch_clause*, so its space is nested inside that of the *exception_specifier*? Or something else? Off hand I'm not sure why introducing two nested spaces per *specific_catch_clause* makes sense…

Mistake? Lack of clarity?

Reply:
My guess is that it's for this:

```csharp
catch (Exception e) when (e is ArgumentException ae)
{
  // Use ae
}
```

We need the declaration space both for the filter (so that the expression can refer to e) and to make sure that `ae` is in scope in the block.

(I know that's a terrible filter, of course...)

Reply:
*exception_specifier* is not listed as introducing a declaration space but as introducing a variable into one. *specific_catch_clause* introduces a declaration space that is shared between the *exception_specifier* and the filter (they don't each need to introduce their own, because the *exception_specifier* wouldn't have a way to observe variables introduced in the filter anyway), and within which the declaration space of the *block* is nested.

Make sense?

Reply:
Well in as much as the design makes sense ;-)